app/src/store/data.ts

```diff
@@ -37,94 +48,288 @@ const lastBlockTimestamp = ref<number>(0);
+    const grantsList = grantsData.grants || [];
+    const grantsDict = grantsList.reduce(
+      (grants: Record<string, string>, grant: Record<string, string>, key: number) => {
+        grants[key.toString()] = grant.payee;
+
+        return grants;
+      },
+      {}
+    );
+    const roundAddresses = grantRoundData.roundAddresses || [];
 
     // Pull state from each GrantRound
-    const grantRoundsList = await Promise.all(
-      roundAddresses.map(async (grantRoundAddress) => {
-        const roundContract = new Contract(grantRoundAddress, GRANT_ROUND_ABI, provider.value);
-        // collect the donationToken before promise.all'ing everything
-        const donationTokenAddress = await roundContract.donationToken();
-        const donationTokenContract = new Contract(donationTokenAddress, ERC20_ABI, provider.value);
-        const matchingTokenAddress = await roundContract.matchingToken();
-        const matchingTokenContract = new Contract(matchingTokenAddress, ERC20_ABI, provider.value);
-
-        return await Promise.all([
-          // round details
-          roundContract.startTime(),
-          roundContract.endTime(),
-          roundContract.metadataAdmin(),
-          roundContract.payoutAdmin(),
-          roundContract.registry(),
-          roundContract.metaPtr(),
-          roundContract.hasPaidOut(),
-          // get donation token details
-          donationTokenContract.name(),
-          donationTokenContract.symbol(),
-          donationTokenContract.decimals(),
-          // get matching token details
-          matchingTokenContract.name(),
-          matchingTokenContract.symbol(),
-          matchingTokenContract.decimals(),
-          matchingTokenContract.balanceOf(grantRoundAddress),
-        ]).then(
-          ([
-            // round details
-            startTime,
-            endTime,
-            metadataAdmin,
-            payoutAdmin,
-            registry,
-            metaPtr,
-            hasPaidOut,
-            // donation token details
-            donationTokenName,
-            donationTokenSymbol,
-            donationTokenDecimals,
-            // matching token details
-            matchingTokenName,
-            matchingTokenSymbol,
-            matchingTokenDecimals,
-            matchingTokenBalance,
-          ]) => {
-            // check for status against `now`
-            const now = Date.now();
+    const grantRoundsList = (await Promise.all(
+      roundAddresses.map(async (grantRoundAddress: string) => {
+        const data = await getGrantRound(grantRoundAddress);
+
+        return data?.grantRound;
+      })
+    )) as GrantRound[];
+
+    // Get latest set of contributions
+    const contributions = await fetchContributions(grantsDict, grantRoundsList[0].donationToken, forceRefresh);
+
+    // Pull all trust scores from gitcoin api
+    const trustBonusScores = await fetchTrustBonusScores(contributions.contributions, forceRefresh);
```

Comment:
Looks like the async calls for `contributions` and `trustBonusScores` can be batched with `Promise.all` to speed it up a bit

Reply:
`TrustBonusScores` depends on the `contributions` to get the `contributor` list